### PR TITLE
[Component]: Alert

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/leo/issues/47

### CODE REVIEW

File: src/components/alert/alert.svelte [hunk 1]
```diff
@@ -0,0 +1,108 @@
+<svelte:options tag="leo-alert" />
+
+<script lang="ts" context="module">
+  export const types = ['info', 'warning', 'error', 'success'] as const
+  export type AlertType = (typeof types)[number]
+
+  export const modes = ['toast', 'callout'] as const
```
Comment (petemill) [inline src/components/alert/alert.svelte:7]: It looks like this differs from the current figma, where toast and callback are only differentiated by background color in light mode. Instead there's `simple` and `full` variations that seem to inform the row vs grid layout. I guess this changed recently @aguscruiz ?

Comment (fallaciousreasoning) [reply src/components/alert/alert.svelte:7]: Nice catch! I've updated this! The colors for dark are actually different with toast/callout too :smile:

File: .storybook/global.svelte [hunk 2]
```diff
@@ -1,7 +1,10 @@
 <script lang="ts">
+  import AlertCentre from '../src/components/alert/alertCentre.svelte'
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply .storybook/global.svelte:2]: Hahaha I can never remember which one is which. American is `Center` then?

[the remark this answers is not in the document]
Comment (AlanBreck) [reply .storybook/global.svelte:2]: Yeah. Doesn't really matter that much. Just made me chuckle. ;-)

[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply .storybook/global.svelte:2]: Renamed :laughing:

**Comment by fallaciousreasoning** [comment]
Hey @aguscruiz, could you give some examples of when designers would use `simple` vs `full` versions of the alert? Also wanted to double check we need some kind of `AlertCenter` component.

**Comment by aguscruiz** [comment]
@fallaciousreasoning 
I don't have anything written down yet, but simple alerts will be used to convey more mundane, sort of short messages (example: "password incorrect"type of deal) 

Whereas full will probably be more of a action-stopper alert. Like something you have to pay attention and probably act upon before continuing the flow.

**Comment by fallaciousreasoning** [comment]
@petemill mind taking another look?

**Review by petemill — APPROVED** [review]
- Secondary buttons might usually sometimes always be Plain-Faint, so we can update example
- Alert center can have toasts move down a little bit (~8px) whilst fading in

**Pull request merged**